src/main/java/net/smartcosmos/dao/objects/domain/ObjectEntity.java

```diff
@@ -24,43 +34,69 @@
 // Need this because Builder will otherwise use the empty constructor
 @AllArgsConstructor(access = AccessLevel.PROTECTED)
 @EntityListeners({ AuditingEntityListener.class })
-@Table(name = "object", uniqueConstraints = @UniqueConstraint(columnNames = { "objecturn",
-        "accountuuid" }) )
+@Table(name = "object", uniqueConstraints = @UniqueConstraint(columnNames = { "objectUrn", "accountUuid" }) )
 public class ObjectEntity implements Serializable {
 
+    private static final int UUID_LENGTH = 16;
+    private static final int OBJECT_URN_LENGTH = 767;
+    private static final int TYPE_LENGTH = 255;
+    private static final int MONIKER_LENGTH = 2048;
+    private static final int NAME_LENGTH = 255;
+    private static final int DESCRIPTION_LENGTH = 1024;
+
+    /*
+        Without setting an appropriate Hibernate naming strategy, the column names specified in the @Column annotations below will be converted
+        from camel case to underscore, e.g.: systemUuid -> system_uuid
+
+        To avoid that, select the "old" naming strategy org.hibernate.cfg.EJB3NamingStrategy in your configuration (smartcosmos-ext-objects-rdao.yml):
+        jpa.hibernate.naming_strategy: org.hibernate.cfg.EJB3NamingStrategy
+     */
+
     @Id
     @GeneratedValue(generator = "uuid2")
     @GenericGenerator(name = "uuid2", strategy = "uuid2")
-    @Column(name = "systemuuid", columnDefinition = "BINARY(16)")
+    @Type(type = "uuid-binary")
+    @Column(name = "systemUuid", length = UUID_LENGTH)
     private UUID id;
 
     @NotNull
-    @Column(name="objecturn", nullable = false, updatable = false)
+    @Size(min = 1, max = OBJECT_URN_LENGTH)
+    @Column(name="objectUrn", length = OBJECT_URN_LENGTH, nullable = false, updatable = false)
     private String objectUrn;
 
     @NotNull
+    @Size(min = 1, max = TYPE_LENGTH)
+    @Column(name = "type", length = TYPE_LENGTH, nullable = false)
     private String type;
 
     @NotNull
-    @Column(name = "accountuuid")
+    @Type(type = "uuid-binary")
+    @Column(name = "accountUuid", length = UUID_LENGTH, nullable = false, updatable = false)
     private UUID accountUrn;
 
     @CreatedDate
     @Column(name = "created", insertable = true, updatable = false)
-    @Temporal(TemporalType.TIMESTAMP)
-    private Date created;
+    private Long created;
```

Comment:
What is the benefit of a `Long` over a `DateTime` other than consistency with the past?  The converters can convert this to/from anything we want.  I think a `DateTime` makes more sense here.